dockstore-client/src/main/java/io/dockstore/client/cli/Client.java

```diff
@@ -1177,6 +1194,256 @@ public static void updateContainerHelp() {
+//            e.printStackTrace();
+        }
+
+        // Pull Dockstore version from matched line
+        Pattern p = Pattern.compile("\"([^\"]*)\"");
+        Matcher m = p.matcher(line);
+        if (m.find()) {
+            currentVersion = m.group(1);
+        }
+
+        return currentVersion;
+    }
+
+    /**
+     * Get the latest stable version name of dockstore available
+     * @return
+         */
+    public static String getLatestVersion() {
+        URL url = null;
+        try {
+            url = new URL("https://api.github.com/repos/ga4gh/dockstore/releases/latest");
+            ObjectMapper mapper = new ObjectMapper();
+            Map<String, Object> map = null;
+            try {
+                map = mapper.readValue(url, Map.class);
+                return map.get("name").toString();
+
+            } catch (IOException e) {
+//                e.printStackTrace();
+            }
+        } catch (MalformedURLException e) {
+//            e.printStackTrace();
+        }
+        return null;
+    }
+
+    /**
+     * Checks if the given tag exists as a release for Dockstore
+     * @param tag
+     * @return
+         */
+    public static Boolean checkIfTagExists(String tag){
+        try {
+            URL url = new URL("https://api.github.com/repos/ga4gh/dockstore/releases");
+            ObjectMapper mapper = new ObjectMapper();
+            try {
+                ArrayList<Map<String,String>> arrayMap = mapper.readValue(url, ArrayList.class);
+                for(Map<String,String> map: arrayMap){
+                    String version = map.get("name");
+                    if(version.equals(tag)) {
+                        return true;
+                    }
+                }
+                return false;
+            } catch (IOException e) {
+//                e.printStackTrace();
+            }
+
+        } catch (MalformedURLException e) {
+//            e.printStackTrace();
+        }
+        return false;
+    }
+
+    /**
+     * Checks for upgrade for Dockstore and install
+     */
+    public static void upgrade(){
+        // Try to get version installed
+        String installLocation = getInstallLocation();
+        if (installLocation == null) {
+            kill("Can't find location of Dockstore executable.  Is it on the PATH?");
+        }
+
+        String currentVersion = getCurrentVersion(installLocation);
+        if (currentVersion == null) {
+            kill("Can't find the current version.");
+        }
+
+        // Update if necessary
+        URL url = null;
+        String latestPath = "https://api.github.com/repos/ga4gh/dockstore/releases/latest";
+        String latestVersion = null;
+        try {
+            url = new URL(latestPath);
+            ObjectMapper mapper = new ObjectMapper();
+            Map<String, Object> map = null;
+            try {
+                // Read JSON from Github
+                map = mapper.readValue(url, Map.class);
```

Comment:
There is a library for talking to git, give it a shot. 

ex: https://github.com/ga4gh/dockstore/blob/783f1ccb95a37e62865d3a00170a65738b639c72/dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java